src/main/java/org/prebid/server/auction/AmpRequestFactory.java

```diff
@@ -125,19 +129,38 @@ private static BidRequest fillExplicitParameters(BidRequest bidRequest, RoutingC
     private static ObjectNode createExtWithDefaults(BidRequest bidRequest, ExtRequestPrebid prebid,
                                                     boolean setDefaultTargeting, boolean setDefaultCache) {
         final boolean isPrebidNull = prebid == null;
+
         return setDefaultTargeting || setDefaultCache
                 ? Json.mapper.valueToTree(ExtBidRequest.of(
                 ExtRequestPrebid.of(
                         isPrebidNull ? Collections.emptyMap() : prebid.getAliases(),
                         isPrebidNull ? Collections.emptyMap() : prebid.getBidadjustmentfactors(),
-                        setDefaultTargeting
-                                ? ExtRequestTargeting.of(Json.mapper.valueToTree(
-                                PriceGranularity.DEFAULT.getBuckets()), true)
-                                : isPrebidNull ? null : prebid.getTargeting(),
+                        createTargetingWithDefaults(prebid),
                         isPrebidNull ? null : prebid.getStoredrequest(),
                         setDefaultCache
                                 ? ExtRequestPrebidCache.of(Json.mapper.createObjectNode())
                                 : isPrebidNull ? null : prebid.getCache())))
                 : bidRequest.getExt();
     }
+
+    /**
+     * Creates updated with default values bidrequest.ext.targeting {@link ExtRequestTargeting} if at least one of it's
+     * child properties is missed or entire targeting does not exist.
+     */
+    private static ExtRequestTargeting createTargetingWithDefaults(ExtRequestPrebid prebid) {
+        final ExtRequestTargeting targeting = prebid != null ? prebid.getTargeting() : null;
+        final boolean isTargetingNull = targeting == null;
+
+        final Boolean includeWinners = isTargetingNull || targeting.getIncludewinners() == null
+                ? true : targeting.getIncludewinners();
+
+        final JsonNode priceGranularity = isTargetingNull ? null : targeting.getPricegranularity();
+        final boolean isPriceGranularityNull = priceGranularity == null || priceGranularity.isNull();
+
+        final JsonNode outgoingPriceGranularityNode = isPriceGranularityNull
+                ? Json.mapper.valueToTree(PriceGranularity.DEFAULT.getBuckets())
+                : priceGranularity;
+
+        return ExtRequestTargeting.of(outgoingPriceGranularityNode, includeWinners);
+    }
```

Comment:
i think the same can be done like:

    private static ExtRequestTargeting createTargetingWithDefaults(ExtRequestPrebid prebid) {
        if (prebid == null || prebid.getTargeting() == null) {
            return ExtRequestTargeting.of(Json.mapper.valueToTree(PriceGranularity.DEFAULT.getBuckets()), true);
        } else {
            final JsonNode priceGranularity = prebid.getTargeting().getPricegranularity();
            final Boolean includeWinners = prebid.getTargeting().getIncludewinners();
            return ExtRequestTargeting.of(priceGranularity.isNull()
                            ? Json.mapper.valueToTree(PriceGranularity.DEFAULT.getBuckets())
                            : priceGranularity,
                    includeWinners == null ? true : includeWinners);
        }
    }